nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/test/java/org/apache/nifi/processors/hadoop/PutHDFSTest.java

```diff
@@ -368,6 +368,52 @@ public void testPutFileWhenDirectoryUsesInvalidEL() {
         runner.assertNotValid();
     }
 
+    @Test
+    public void testPutFilePermissions() throws FileNotFoundException, IOException{
```

Comment:
Could you please break this method into three separate tests?
* testPutFilePermissionsWithProcessorConfiguredUmask
* testPutFilePermissionsWithXmlConfiguredUmask
* testPutFilePermissionsWithNoConfiguredUmask